### ISSUE TITLE
PropsSI does not accept arrays as inputs (MATLAB)

### ISSUE THREAD

**Issue by dpokras** [CoolProp/CoolProp#2220]
Do you want to ask a question? Are you looking for technical support? For any kind of question regarding CoolProp and its usage, you can ask the [CoolProp user group](https://goo.gl/Pa7FBT)

If you have an issue to report, please continue and fill out the applicable sections below.  The details provided will help to resolve this issue as quickly as possible.

### Prerequisites

Before posting an issue, please:  
    * Check the [CoolProp User Group](https://goo.gl/Pa7FBT) for common solutions.  
    * Check that your issue isn't already filed under [CoolProp Issues](https://github.com/CoolProp/CoolProp/issues).  
    * Check the CoolProp [latest release](http://www.coolprop.org) and [development version](http://www.coolprop.org/dev) documentation.  

### Description

I am unable to put an array as an input for PropsSI in the MATLAB wrapper. Only accepts 'real numbers.'

### Steps to Reproduce

I used this example code from your repository:
demoCase = cell(6,1);
demoCase{1} = {'V','D'};      % Desired outputs
demoCase{2} = 'T';            % 1st input name
demoCase{3} = 200:10:6000;    % 1st input value(s)
demoCase{4} = 'P';            % 2nd input name
demoCase{5} = 5000:100:20000; % 2nd input value(s)
demoCase{6} = 'Argon';        % Fluid name
props = py.CoolProp.CoolProp.PropsSI(demoCase{:});

**Expected behavior:** 

I expected an array as an output.

**Actual behavior:** 

Error:

Error using CoolProp/CoolProp.PropsSI
Python Error: TypeError: must be real number, not
array.array

### Versions

**CoolProp Version:** CoolProp-6.4.3.post1 
**Operating System and Version:** Windows 11
**Access Method:** MATLAB 9.13.0.2105380 (R2022b) Update 2 / python 3.8

### Additional Information

If possible, please post examples and/or screenshots of the issue.

![Screenshot 2023-03-06 174013](https://user-images.githubusercontent.com/105276214/223174674-3c772ab1-2150-4436-b9f2-2868c3745242.jpg)

**Comment by ibell** [comment]
That's right, but the true Python interface does. I'm frankly surprised
that the array inputs are not automatically converted.

**Comment by dpokras** [comment]
Hi Ian,

Is there anything I could do from my side?

Best,
Daniel

**Comment by ibell** [comment]
You need to figure out how MATLAB goes the conversion; it seems it is converting to ``array.array`` but you want it to convert to ``numpy.array``.

**Comment by dpokras** [comment]
I understood a solution to the input issue. CoolProp doesn't accept array.array inputs but will instead take structures and convert them to np.arrays. I would suggest that an error message include this answer, otherwise would it be possible to make an implementation to allow for array.array inputs.

**Comment by ibell** [comment]
Nope, this is a MATLAB "problem"; it should be converting to numpy arrays. Nothing we can do in CoolProp land. Already handling too many array conversion cases

**Comment by HenningScheufler** [comment]
Is there a work around?

**Comment by ibell** [comment]
Only thing I can think of is to use python

**Comment by Arnab-Mahanti** [comment]
Not Very robust but I created a wrapper for y personal use. Maybe useful.

```MATLAB
function result=PropsSI(varargin)
        if nargin==2
        result=py.CoolProp.CoolProp.PropsSI(varargin{:});
        elseif nargin==6
            if isequal(size(varargin{5}),[1 1])
                varargin{5}=varargin{5}.*ones(size(varargin{3}));
                result=arrayfun(@(a,b)(py.CoolProp.CoolProp.PropsSI(varargin{1:2},a,varargin{4},b,varargin{6})),varargin{3},varargin{5});
            elseif isequal(size(varargin{5}),size(varargin{3}))
                result=arrayfun(@(a,b)(py.CoolProp.CoolProp.PropsSI(varargin{1:2},a,varargin{4},b,varargin{6})),varargin{3},varargin{5});
            else
                error("Incorrect sizes of arguments in PropsSI");
            end
        else
            error("Incorrect number of arguments in PropsSI");
        end
end
```

**Comment by PerplexedFox** [comment]
Convert the values to np.array. 
Works perfectly for me
`py.CoolProp.CoolProp.PropsSI('T','P',py.numpy.array(p),'D',**py.numpy.array(rho)**,"Nitrogen");`
